src/WebJobs.Extensions.SendGrid/Bindings/SendGridHelpers.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            // Apply message defaulting
+            if (mail.From == null)
+            {
+                if (!string.IsNullOrEmpty(attribute.From))
+                {
+                    EmailAddress from = null;
+                    if (!TryParseAddress(attribute.From, out from))
+                    {
+                        throw new ArgumentException("Invalid 'From' address specified");
+                    }
+                    mail.From = from;
+                }
+                else if (config.FromAddress != null)
+                {
+                    mail.From = config.FromAddress;
+                }
+            }
+
+            if (!IsToValid(mail))
+            {
+                if (!string.IsNullOrEmpty(attribute.To))
+                {
+                    EmailAddress to = null;
+                    if (!TryParseAddress(attribute.To, out to))
+                    {
+                        throw new ArgumentException("Invalid 'To' address specified");
+                    }
+
+                    mail.AddTo(to);
+                }
+                else if (config.ToAddress != null)
+                {
+                    mail.AddTo(config.ToAddress);
+                }
+            }
+
+            if (string.IsNullOrEmpty(mail.Subject) &&
+                !string.IsNullOrEmpty(attribute.Subject))
+            {
+                mail.Subject = attribute.Subject;
+            }
+
+            if ((mail.Contents == null || mail.Contents.Count == 0) &&
+                !string.IsNullOrEmpty(attribute.Text))
+            {
+                mail.AddContent("text/plain", attribute.Text);
+            }
+        }
+
+        internal static SendGridMessage CreateMessage(string input)
+        {
+            JObject json = JObject.Parse(input);
+            return CreateMessage(json);
+        }
+
+        internal static SendGridMessage CreateMessage(JObject input)
+        {
+            return input.ToObject<SendGridMessage>();
+        }
+
+        internal static string CreateString(SendGridMessage input)
+        {
+            return CreateString(JObject.FromObject(input));
+        }
+
+        internal static string CreateString(JObject input)
+        {
+            return input.ToString(Formatting.None);
+        }
+
+        internal static bool IsToValid(SendGridMessage item)
+        {
+            return item.Personalizations != null &&
+                item.Personalizations.Count > 0 &&
+                item.Personalizations.All(p => p.Tos != null && !p.Tos.Any(t => string.IsNullOrEmpty(t.Email)));
+        }
+
+        internal static SendGridConfiguration CreateConfiguration(JObject metadata)
+        {
+            SendGridConfiguration sendGridConfig = new SendGridConfiguration();
+
+            JObject configSection = (JObject)metadata.GetValue("sendGrid", StringComparison.OrdinalIgnoreCase);
```

Comment:
I think the ApplyConfig utility might help here - I think we should allow the properties `fromAddress` and `toAddress` so that there's not a disconnect between host.json and config values (while still supporting the old to and from values)